conda-store-server/conda_store_server/__init__.py

```diff
@@ -46,10 +49,28 @@ def _version2_fmt(build: "Build") -> str:  # noqa: F821
         name = build.specification.name
         return f"{hash}-{timestamp}-{id}-{name}"
 
+    def _version3_fmt(build: "Build") -> str:  # noqa: F821
+        # Adds namespace here to separate builds performed by different users
+        # since all builds are stored in the same directory in v3, see
+        # Build.build_path in orm.py. Additionally, this also hashes the
+        # timestamp and build id just to make collisions very unlikely
```

Comment:
Do we want to hash all these components? My first impression is that hashing the lockfile contents would be enough. If that hash is the same for two environments, then why recreate them, right?

If we consider ACLs and concurrent execution from different users, I guess I can see why we add `namespace` to the mix. Security, yadda yadda. However, why do we want to prevent collisions with timestamp and build_id? Can't we just say "oh, this thing already exists in your namespace, there you go" and save a few bytes and inodes?

Reply:
I feel this discussion should involve more members of the team and shouldn't be based on implementation details alone. Pinging @conda-incubator/conda-store for awareness.

Reply:
Okay, we can wait until the next team meeting.